### DISCUSSION TITLE
The uuid is not generated with the HasUuids trait when extending the roles and permissions base class

### DISCUSSION THREAD

**Discussion by ginogalarza** [spatie/laravel-permission#2620] [Q&A] [answered]
Hello, I want to use uuid for the primary keys of the roles and permissions models, I followed the steps in the documentation to use these, with the difference that I keep the name id as the primary key. I configured everything as indicated, but when creating a new role or permissions record I get a sql error `1364 Field 'id' doesn't have a default value`. Thanks for the help. 

**Configuration of my models:**

```
class Permission extends SpatiePermission
{
    use HasFactory;
    use HasUuids;
}
```

![image](https://github.com/spatie/laravel-permission/assets/27784870/031e4f38-484d-403c-b95a-da3233fc3be0)

**Comment by drbyte** [comment, marked as answer]
> but when creating a new role or permissions record I get a sql error 1364 Field 'id' doesn't have a default value. 

What's your code for creating the record? Including imported classes.

**Reply by ginogalarza** [reply]
Thanks for the help, I hadn't realized that I was making the importance of the base classes and not the custom ones
